frontend/kesaseteli/employer/src/components/application/steps/step1/EmployerForm.tsx

```diff
@@ -0,0 +1,72 @@
+import CompanyInfoGrid from 'kesaseteli/employer/components/application/companyInfo/CompanyInfo';
+import Checkbox from 'kesaseteli/employer/components/application/form/Checkbox';
+import TextInput from 'kesaseteli/employer/components/application/form/TextInput';
+import ContactInputs from 'kesaseteli/employer/components/application/steps/step1/ContactInputs';
+import useApplicationApi from 'kesaseteli/employer/hooks/application/useApplicationApi';
+import { useTranslation } from 'next-i18next';
+import React from 'react';
+import { useFormContext } from 'react-hook-form';
+import FormSection from 'shared/components/forms/section/FormSection';
+import useMountEffect from 'shared/hooks/useMountEffect';
+import Application from 'shared/types/application-form-data';
+
+const EmployerForm: React.FC = () => {
+  const { t } = useTranslation();
+  const stepTitle = t('common:application.step1.header');
+
+  const { application } = useApplicationApi();
+
+  const showInitially = application?.is_separate_invoicer || false;
+  const [showInvoicer, setShowInvoicer] =
+    React.useState<boolean>(showInitially);
+
+  useMountEffect(() => {
+    setShowInvoicer(showInitially);
+  });
+
+  const { getValues } = useFormContext<Application>();
+
+  const { updateApplication } = useApplicationApi();
+
+  const handleInvoicerCheckboxChange = React.useCallback(
+    (value: boolean) => {
+      setShowInvoicer(value);
+      if (!value) {
+        updateApplication({
+          ...getValues(),
+          invoicer_name: '',
+          invoicer_email: '',
+          invoicer_phone_number: '',
+        });
+      }
+    },
+    [setShowInvoicer, updateApplication, getValues]
+  );
+
+  return (
+    <>
+      <FormSection
+        header={stepTitle}
+        tooltip={t('common:application.step1.tooltip')}
+        withoutDivider
+      />
+      <CompanyInfoGrid />
+      <FormSection columns={2}>
+        <ContactInputs type="contact_person" />
+        <TextInput
+          id="street_address"
+          validation={{ required: true, maxLength: 256 }}
+        />
+        <Checkbox
+          $colSpan={2}
+          id="is_separate_invoicer"
```

Comment:
Not sure if you're using these ids anywhere else. If so, it's worth extracting them as constants.

Reply:
Yes id is actually form field name and typed, not just string: https://react-hook-form.com/api/useform/register
If you look into Checkbox.tsx you'll see that the id is registered through the `useFormContext<Application>`-hook, so it must be a valid field path of `Application` type. In my opinion contants does not help here but adds just unnecessary complexity. If you typo this name like `isss_separate_invoicer`, the typescript will catch it.